### PR TITLE
Ops 2870/group and extra checks for api endpoints

### PR DESCRIPTION
**Pull request by Santi-3rd**
## What changed

Removed group and extra checks from authorization_providers.py. Made needed changes to endpoints and tests. 

## Issue

#2870 

## How to test

Run backend tests.

## Definition of Done Checklist
- [x] OESA: Code refactored for clarity
- [x] OESA: Dependency rules followed
- [x] Automated unit tests updated and passed
- [x] Automated integration tests updated and passed
- [x] Automated quality tests updated and passed
- [x] Automated load tests updated and passed
- [x] Automated a11y tests updated and passed
- [x] Automated security tests updated and passed
- [x] 90%+ Code coverage achieved
- [x] Form validations updated

### CODE REVIEW

**Review by rajohnson90 — APPROVED** [review]

File: backend/ops_api/ops/resources/procurement_steps.py [hunk 1]
```diff
@@ -52,42 +50,6 @@ def get_current_user_id():
 
 
 # TODO: considering refactoring to DRYer along with similar code in services_component.py and budget_line_items.py
```
Comment (fpigeonjr) [inline backend/ops_api/ops/resources/procurement_steps.py:52]: was this todo included with this work?

**Review by fpigeonjr — COMMENTED** [review]
🤩 looks good, but will defer to others

File: backend/ops_api/ops/resources/procurement_steps.py (hunk 1 above)
Comment (fpigeonjr) [reply backend/ops_api/ops/resources/procurement_steps.py:52]: ok if the work is no longer TODO, maybe we can remove the comment

**Review by maiyerlee — APPROVED** [review]
🎉🎉🎉

File: backend/data_tools/data/user_data.json5 [hunk 2]
```diff
@@ -50,6 +50,8 @@
         "PUT_BUDGET_LINE_ITEM",
         "PATCH_BUDGET_LINE_ITEM",
         "POST_BUDGET_LINE_ITEM",
+        // Delete permissions are only for local, not for prod. Users should not have delete BLI permissions in prod.
```
Comment (jonnalley) [inline backend/data_tools/data/user_data.json5:53]: @Santi-3rd @rajohnson90 what’s the right long term solution here ?  Granting extra permissions just to appease tests is problematic

**Pull request merged**